src/main/kotlin/com/criptext/mail/scenes/emaildetail/workers/ReadEmailsWorker.kt

```diff
@@ -33,13 +33,14 @@ class ReadEmailsWorker(private val dao: EmailDao,
 
     override fun work(reporter: ProgressReporter<EmailDetailResult.ReadEmails>)
             : EmailDetailResult.ReadEmails? {
-        val threadIds = dao.getThreadIdsFromEmailIds(emailIds)
+        val emails = dao.getAllEmailsByMetadataKey(metadataKeys)
+        val unreadEmails = emails.filter { it.unread }
         val result = Result.of { apiClient.postOpenEvent(metadataKeys) }
-                .map { apiClient.postThreadReadChangedEvent(threadIds,false) }
-                .flatMap { Result.of { dao.toggleRead(ids = emailIds, unread = false) } }
+                .map { apiClient.postThreadReadChangedEvent(unreadEmails.map { it.threadId }.distinct(),false) }
```

Comment:
this should use `flatMap` since api calls are error prone.

Reply:
Ok